services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/service/connected/ConnectedDevicesService.java

```diff
@@ -57,43 +61,66 @@ public Command call() {
         };
     }
 
-    private LldpPacket deserialize(Ethernet eth) {
-        if (eth.getPayload() instanceof LLDP) {
-            return new LldpPacket(eth.getPayload().serialize());
+    private LldpPacket deserializeLldp(Ethernet eth, SwitchId switchId, long cookie) {
+        try {
+            if (eth.getPayload() instanceof LLDP) {
+                return new LldpPacket(eth.getPayload().serialize());
+            }
+        }  catch (Exception exception) {
+            logger.info("Could not deserialize LLDP packet {} on switch {}. Cookie {}. Deserialization failure: {}, "
+                    + "exception: {}", eth, switchId, cookie, exception.getMessage(), exception);
+            return null;
         }
+        logger.info("Got invalid lldp packet: {} on switch {}. Cookie {}", eth, switchId, cookie);
         return null;
     }
 
+    private boolean isLldpRelated(long value) {
+        return value == LLDP_INPUT_PRE_DROP_COOKIE
+                || value == LLDP_TRANSIT_COOKIE
+                || Cookie.isFlowLldp(value);
+    }
+
     private void handlePacketIn(OfInput input) {
         U64 rawCookie = input.packetInCookie();
 
-        if (rawCookie == null || !Cookie.isMaskedAsLldp(rawCookie.getValue())) {
+        if (rawCookie == null || !isLldpRelated(rawCookie.getValue())) {
             return;
         }
 
         long cookie = rawCookie.getValue();
         SwitchId switchId = new SwitchId(input.getDpId().getLong());
 
-        LldpPacket lldpPacket;
-        try {
-            lldpPacket = deserialize(input.getPacketInPayload());
-            if (lldpPacket == null) {
-                logger.info("Got invalid lldp packet: {} on switch {}. Cookie {}",
-                        input.getPacketInPayload(), switchId, cookie);
-                return;
-            }
-        } catch (Exception exception) {
-            logger.info("Could not deserialize LLDP packet {} on switch {}. Cookie {}. Deserialization failure: {}, "
-                    + "exception: {}", input.getPacketInPayload(), switchId, cookie, exception.getMessage(), exception);
+        if (Cookie.isFlowLldp(cookie)) {
+            handleFlowLldp(input, switchId, cookie);
+        } else if (cookie == LLDP_TRANSIT_COOKIE || cookie == LLDP_INPUT_PRE_DROP_COOKIE) {
```

Comment:
Is it the same as isLldpRelated checks?

Reply:
no. there is no `Cookie.isFlowLldp(value)`

isLldpRelated is "sum" of cases form `if` and from `else`